core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -922,6 +922,18 @@ object Chunk
     */
   final class Queue[+O] private (val chunks: SQueue[Chunk[O]], val size: Int) extends Chunk[O] {
 
+    private[this] lazy val accumulatedLengths: Array[(Int, Chunk[O])] = {
+      val arr = new Array[(Int, Chunk[O])](chunks.size)
+      var accLen = 0
+      var i = 0
+      chunks.foreach { c =>
+        accLen += c.size
+        arr(i) = (accLen, c)
+        i += 1
+      }
+      arr
+    }
+
```

Comment:
So, there could be some memory svings in returning a pair of arrays, rather than an array of pairs. First, that would avoid allocating the tuple objects. Second, it would allow boxing, and thus dereferencing, the integers.

```suggestion
    private[this] lazy val accumulatedLengths: (Array[Int], Array[Chunk[O]]) = {
      val lens = new Array[Int](chunks.size)
      val arr   = new Array[Chunk[O]](chunks.size)
      var accLen = 0
      var i = 0
      chunks.foreach { c =>
        accLen += c.size
        lens(i) = accLen
        arr(i) = c
        i += 1
      }
      (lens, arr)
    }

```

Reply:
Sounds good, will make the change